File: 프로그래머스/0/120811. 중앙값 구하기/중앙값 구하기.java
class Solution {
    public int solution(int[] array) {
        int answer = 0;
        
        for(int i = 0; i < array.length - 1; i++){ //버블정렬을 사용하여 오름차순으로 정렬
            for(int j = 0; j < array.length - 1 - i; j++){
                if(array[j] > array[j + 1]){ //원소 위치 바꾸기
                    int num = array[j]; //원소를 비교하여 오름차순으로 정렬하기 위한 변수
                    array[j] = array[j + 1];
                    array[j + 1] = num;
                }
            }
        }
        return array[(array.length-1)/2]; //정렬된 배열의 중간값을 반환
    }
}

File: 프로그래머스/0/120811. 중앙값 구하기/중앙값 구하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}
// array_len은 배열 array의 길이입니다.
int solution(int array[], size_t array_len) {
    int answer = 0;
    qsort(array, array_len, sizeof(int), compare);
    if (array_len % 2 == 1) { // 배열의 길이가 홀수인 경우
        return (double)array[array_len/2];
    } else { // 배열의 길이가 짝수인 경우
        return (double)(array[array_len/2-1] + array[array_len/2])/2;
    }
    return answer;
}